iModelJsNodeAddon/presentation/ECPresentationSerializer.cpp

```diff
@@ -907,7 +917,11 @@ NavNodeKeyPtr IModelJsECPresentationSerializer::_GetNavNodeKeyFromJson(IConnecti
 NavNodeKeyPtr IModelJsECPresentationSerializer::_GetBaseNavNodeKeyFromJson(BeJsConst json) const
     {
     Utf8CP type = json["type"].asCString();
-    return NavNodeKey::Create(type, "", ParseNodeKeyHashPath(json["pathFromRoot"]));
+    NavNodeKeyPtr key = NavNodeKey::Create(type, "", ParseNodeKeyHashPath(json["pathFromRoot"]));
+    std::unique_ptr<PresentationQuery> instanceKeysSelectQuery = GetPresentationQueryFromJson(json["instanceKeysSelectQuery"]);
+    if (!instanceKeysSelectQuery->GetQueryString().empty())
+        key->SetInstanceKeysSelectQuery(std::move(instanceKeysSelectQuery));
+    return key;

```

Comment:
let's change `GetPresentationQueryFromJson` to return `nullptr` when the query is invalid. then we can just do this:
```cpp
key->SetInstanceKeysSelectQuery(GetPresentationQueryFromJson(json["instanceKeysSelectQuery"]));
```